yt_dlp/extractor/archiveorg.py

```diff
@@ -241,3 +256,153 @@ def _real_extract(self, url):
                     'parent': 'root'})
 
         return info
+
+
+class YoutubeWebArchiveIE(InfoExtractor):
+    IE_NAME = 'youtube:web.archive.org'
+    IE_DESC = 'web.archive.org saved youtube videos'
+    _VALID_URL = r"""(?x)^
+                (?:https?://)?web\.archive\.org/
+                    (?:web/)?
+                    (?:[0-9A-Za-z_*]+/)?  # /web and the version index is optional
+                (?:https?://)?
+                (?:
+                    (?:\w+\.)?youtube\.com/watch\?v=  # Youtube URL
+                    |(wayback-fakeurl\.archive\.org/yt/)  # Or optionally, also support the internal fake url
+                )
+                (?P<id>[0-9A-Za-z_-]{11})(?:\#|&|$)
```

Comment:
```suggestion

                (?:https?(?::|%3[Aa])//)?
                (?:
                    (?:\w+\.)?youtube\.com/watch(?:\?|%3[fF])(?:[^\#]+(?:&|%26))?v(?:=|%3[dD])  # Youtube URL
                    |(wayback-fakeurl\.archive\.org/yt/)  # Or the internal fake url
                )
                (?P<id>[0-9A-Za-z_-]{11})(?:%26|\#|&|$)
```

Here are a few test cases:
```
https://web.archive.org/web/20120712231619/http://www.youtube.com/watch?gl=US&v=AkhihxRKcrs&hl=en
https://web.archive.org/web/20120712231619/http%3A//www.youtube.com/watch%3Fv%3DAkhihxRKcrs%26gl%3DUS%26hl%3Den
https://web.archive.org/web/20120712231619/http%3A//www.youtube.com/watch%3Fgl%3DUS%26v%3DAkhihxRKcrs%26hl%3Den
```